### PR TITLE
Add a new person filter rule 'HasAddressText'

### PR DESCRIPTION
**Pull request by Mattkmmr**
Allows users to filter persons who have an address containing a given text.
Fixes [#9168](https://gramps-project.org/bugs/view.php?id=9168)

### CODE REVIEW

File: gramps/gen/filters/rules/person/_hasaddresstext.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+# -------------------------------------------------------------------------
+#
+# HasAddressText
+#
+# -------------------------------------------------------------------------
+class HasAddressText(Rule):
+    """Rule that checks for a string in personal addresses"""
+
+    labels = [_('Adress:')]
```
Comment (Nick-Hall) [inline gramps/gen/filters/rules/person/_hasaddresstext.py]: Please fix spelling.

**Comment by prculley** [comment]
Are you aware that the rule "People with records containing substring" already searches in the person's addresses?  Or did you add this because it is faster than the more general rule?

**Comment by Mattkmmr** [comment]
My thinking was that the "People with records containing substring" rule gives back many results which mached other fields than adresses, especially when you don't know the exact adress. Then you have to edit/open each person to check if the adress was matched or something else, which could be avoided if only the adress fields were checked in a new filter.

**Comment by Nick-Hall** [comment]
Yes, it seems that a separate rule may be useful.

**Pull request merged**